proposals/shared-everything-threads/Overview.md

```diff
@@ -498,29 +502,32 @@ behave mostly like [sealed] JS objects, and their identities survive round trips
 across thread boundaries. Additionally, to facilitate calling methods on these shared objects, they
 have [realm-local prototypes][tls-prototypes] just like shared structs are proposed to have in JS.
 
-The shared versions of the JS API types (besides memories) will be represented with new
-constructors:
+The shared versions of the JS API types will share constructors with their existing unshared
+versions, but will take an additional `shared: bool` option in an options bag argument. These
```

Comment:
The shared attribute is part of the Wasm function/table/... type, and hence will also have to be an (optional) property of the type descriptor returned by JS type reflection. One of the design goals of the JS types proposal was that all the constructors symmetrically take a corresponding type descriptor as argument. Hence, the `shared` attribute should not be in a random options bag but go inside the existing type descriptor.

Reply:
cc @SPY, is it going to be easy to extend the type reflection API with this extra sharedness information?

Reply:
Yes, I don't see any problem to extend signature with `shared` attribute to keep symmetry.